modules/starfish/2.0/starfish2.smk

```diff
@@ -0,0 +1,247 @@
+        tbi = CFG["dirs"]["inputs"] + "vcf/{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}.{caller}.vcf.gz.tbi"
+    run:
+        op.relative_symlink(input.vcf, output.vcf), 
+        op.relative_symlink(input.vcf + ".tbi", output.tbi)
+
+
+# Run Starfish
+rule _starfish_run:
+    input:
+        vcfs = expand(
+            CFG["dirs"]["inputs"] + "vcf/{{seq_type}}--{{genome_build}}/{{tumour_id}}--{{normal_id}}--{{pair_status}}.{caller}.vcf.gz", 
+            caller = callers
+            ),
+        tbis = expand(
+            CFG["dirs"]["inputs"] + "vcf/{{seq_type}}--{{genome_build}}/{{tumour_id}}--{{normal_id}}--{{pair_status}}.{caller}.vcf.gz.tbi", 
+            caller = callers
+            ),
+        reference = reference_files("genomes/{genome_build}/sdf"),
+        starfish_script = CFG["inputs"]["starfish_script"]
+    output:
+        complete = touch(CFG["dirs"]["starfish"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/starfish.complete"), 
+        venn = CFG["dirs"]["starfish"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/venn.pdf"
+    log:
+        stdout = CFG["logs"]["starfish"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/starfish_run.stdout.log",
+        stderr = CFG["logs"]["starfish"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/starfish_run.stderr.log"
+    params:
+        opts = CFG["options"]["starfish_run"]
+    conda:
+        CFG["conda_envs"]["starfish"]
+    threads:
+        CFG["threads"]["starfish_run"]
+    resources:
+        **CFG["resources"]["starfish_run"]
+    shell:
+        op.as_one_line("""
+        {input.starfish_script} --sdf {input.reference} -O $(dirname {output.complete})
+        --names {callers} --threads {threads}
+        {params.opts} --venn --verbose
+        -V {input.vcfs} > {log.stdout} 2> {log.stderr} 
+        """)
+
+checkpoint _starfish_rename_output: 
+    input: 
+        complete = CFG["dirs"]["starfish"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/starfish.complete"
+    output: 
+        renamed = touch(CFG["dirs"]["starfish"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/starfish.renamed")
+    run: 
+        # Get the starfish letter names for the variant callers
+        names = list(string.ascii_uppercase[0:(len(callers))])
+        # Generate a dictionary of letter names and variant callers
+        name_dict = {names[i]: callers[i] for i in range(len(callers))}
+        outdir = os.path.dirname(input.complete)
+        for src in glob.glob(outdir + "/[A-Z]*.vcf.gz*"): 
+            bname = os.path.basename(src)
+            lhs, rhs = bname.split(".", 1)
+            lhs = "_".join(lhs)
+            for key in list(name_dict.keys()): 
+                lhs = lhs.replace(key, name_dict[key])
+            dest = os.path.join(outdir, ".".join([lhs, rhs]))
+            os.rename(src, dest) 
```

Comment:
If you are still getting that tabix index error. This might be a good place to just "touch" all the .tbi files to make sure they are newer than their VCF